### DISCUSSION TITLE
Is there a shortcut to home directory in scripts? (MacOS)

### DISCUSSION THREAD

**Discussion by benjamin-guibert** [johnlindquist/kit#1140] [Q&A] [answered]
I noticed that using the home directory from a path with `~` or `$HOME` is not working when using file system (_MacOS_). I can only manage to use absolute path (`/Users/...`).

Is there another way? or should I declare an environment variable?

With the following example, I get `Error: ENOENT: no such file or directory, scandir '~/Downloads'` whereas with `/Users/...`, it works.
```typescript
import '@johnlindquist/kit'

import { readdirSync } from "fs"

// The following doesn't work:
const dir = readdirSync('~/Downloads')
// The following works:
// const dir = readdirSync('Users/xxx/Downloads')

log(dir)
```

**Comment by johnlindquist** [comment, marked as answer]
@benjamin-guibert 

Try this:
```js
const dir = home("Downloads", "some-dir", "my-file.txt")
```

Using `home` enables cross-platform/user/etc support

**Reply by benjamin-guibert** [reply]
Great, thanks! 🙏🏻
